### ISSUE TITLE
[DUW] Update year of discharge dropdown to an input field

### ISSUE THREAD

**Issue by thejordanwood** [department-of-veterans-affairs/va.gov-cms#20089]
## Description

The year dropdown for "What year were you discharged from the military?" should be changed to [an input field](https://design.va.gov/components/form/text-input). 

VA Design System guidelines suggest considering a different component when there are more than 15 options in the dropdown. [Guidelines for select component](https://design.va.gov/components/form/select)

**Existing design:**

<img width="686" alt="Image" src="https://github.com/user-attachments/assets/9a04920a-49bd-4e6a-8c8e-63d61c911cf3" />

### User story

**AS A** Veteran using the Discharge Upgrade Wizard and answering what year I was discharged
**I WANT** to type the year discharged instead of scrolling in a dropdown
**SO THAT**  I can complete it with the least amount of angst and effort, 

#### Error message to use

![Image](https://github.com/user-attachments/assets/1f7ff049-69f3-489f-9fe9-b4336a63045a)

## Acceptance criteria

- [ ] The question  "What year were you discharged from the military? is now an [an input field](https://design.va.gov/components/form/text-input). 
- [ ] When user enters the year, all existing logic and functionality works as expected
- [ ] When user enters a year that is  current year - 15, the month question then displays.
- [ ] When user enters a year that is 1991 or before, then the logic that is currently used for Before 1992 works as expected.
- [ ] If user enters anything other than numbers, an error message displays in red  'Must be four digits.' and the existing red indicators still display (see screenshot above, but note that we are using digits and not characters.)'
- [ ] If user enters less or more than four characters, an error message displays in red ~~Must be four digits.~~ 'Enter a valid 4 digit year', and the existing red indicators still display (see screenshot above, but note that we are using digits and not characters.)'
- [ ] If user enters a year in the future, an error message displays in red ~~Year can not be in the future~~ 'Enter a valid 4 digit year', and the existing red indicators still display (see screenshot above) 
- [ ] Does this product have an existing regression test plan?
	- [x] Yes, link [here](https://dvagov.sharepoint.com/:x:/s/SitewidePublicWebsites/EaqLbCh8zzxHszmtMem6hjoBXW7quKAoKUDevCOuQaadjA?email=Ian.Harrison%40va.gov&e=xw8y6x)
- [x] Regression test updated by @FranECross 
- [ ] Design review
- [ ] Accessibility review

**Comment by FranECross** [comment]
@FranECross Think about whether we want to restrict users from entering a very old year.

**Comment by chriskim2311** [comment]
Updated guidance on error message from Jordan
https://github.com/department-of-veterans-affairs/vets-website/pull/34087#issuecomment-2594018006

**Comment by FranECross** [comment]
Thanks, @chriskim2311 and Jordan! I updated the AC.

**Comment by FranECross** [comment]
@chriskim2311 I missed this one yesterday when testing DUW tickets. I'll look at it today.

**Comment by jilladams** [comment]
This isn't deployed to prod yet -- it'll go today in the vets-website deploy.

**Comment by jilladams** [comment]
(It _is_ present in staging.)

**Comment by FranECross** [comment]
@thejordanwood Question: When a Veteran enters 1991 or before in the year of discharge, the Results screen still displays "... before 1992", instead of the actual year they entered in the text field. Shouldn't this display the year so as not to confuse them? Thanks. cc @chriskim2311

**Comment by thejordanwood** [comment]
@FranECross Good catch! I agree that it should display the specific year of discharge.

**Comment by jilladams** [comment]
The year as an input is complete in prod. @chriskim2311 if adjusting handling of that Results year is ok to do here, let's do, but if a separate ticket is needed, holler.
